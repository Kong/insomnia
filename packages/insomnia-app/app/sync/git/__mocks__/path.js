// eslint-disable-next-line filenames/match-exported
const path = jest.requireActual('path');

const exportObj = { __mockPath, ...path };

function __mockPath(type) {
  const mock = type === 'win32' ? path.win32 : path.posix;

  Object.keys(mock).forEach(k => (exportObj[k] = mock[k]));
}

module.exports = exportObj;
