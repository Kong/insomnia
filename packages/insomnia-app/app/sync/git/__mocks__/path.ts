// eslint-disable-next-line filenames/match-exported
const path = jest.requireActual('path');

const exportObj = { __mockPath, ...path };

function __mockPath(type) {
  const mock = type === 'win32' ? path.win32 : path.posix;

  Object.keys(mock).forEach(k => {
    exportObj[k] = mock[k];
  });
}

// WARNING: changing this to `export default` will break the mock and be incredibly hard to debug. Ask me how I know.
module.exports = exportObj;
