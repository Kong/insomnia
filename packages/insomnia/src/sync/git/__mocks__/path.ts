import { jest } from '@jest/globals';
import pathOriginal, { PlatformPath } from 'path';

// eslint-disable-next-line filenames/match-exported
const path = jest.requireActual('path') as typeof pathOriginal;

const exportObj = { __mockPath, ...path } as any;

function __mockPath(type: string) {
  const mock = type === 'win32' ? path.win32 : path.posix;

  Object.keys(mock).forEach(k => {
    exportObj[k] = mock[k as keyof PlatformPath];
  });
}

// WARNING: changing this to `export default` will break the mock and be incredibly hard to debug. Ask me how I know.
module.exports = exportObj;
