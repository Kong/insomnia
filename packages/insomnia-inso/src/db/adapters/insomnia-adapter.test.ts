import path from 'path';
import insomniaAdapter from './insomnia-adapter';

describe('insomniaAdapter()', () => {
  const fixturesPath = 'src/db/fixtures';

  it('should seed with data file (JSON)', async () => {
    const pathname = path.join(fixturesPath, 'insomnia-v4', 'insomnia_v4.json');
    const db = await insomniaAdapter(pathname);

    expect(db?.ApiSpec.length).toBe(1);
    expect(db?.Environment.length).toBe(2);
    expect(db?.Request.length).toBe(13);
    expect(db?.RequestGroup.length).toBe(1);
    expect(db?.Workspace.length).toBe(1);
    expect(db?.UnitTestSuite.length).toBe(1);
    expect(db?.UnitTest.length).toBe(2);
  });

  it('should seed with data file and its filters (JSON)', async () => {
    const pathname = path.join(fixturesPath, 'insomnia-v4', 'insomnia_v4.json');
    const db = await insomniaAdapter(pathname, ['Environment']);

    expect(db?.ApiSpec.length).toBe(0);
    expect(db?.Environment.length).toBe(2);
    expect(db?.Request.length).toBe(0);
    expect(db?.RequestGroup.length).toBe(0);
    expect(db?.Workspace.length).toBe(0);
    expect(db?.UnitTestSuite.length).toBe(0);
    expect(db?.UnitTest.length).toBe(0);
  });

  it('should seed with data file (YAML)', async () => {
    const pathname = path.join(fixturesPath, 'insomnia-v4', 'insomnia_v4.yaml');
    const db = await insomniaAdapter(pathname);

    expect(db?.ApiSpec.length).toBe(1);
    expect(db?.Environment.length).toBe(2);
    expect(db?.Request.length).toBe(13);
    expect(db?.RequestGroup.length).toBe(1);
    expect(db?.Workspace.length).toBe(1);
    expect(db?.UnitTestSuite.length).toBe(1);
    expect(db?.UnitTest.length).toBe(2);
  });

  it('should seed with data file and its filters (YAML)', async () => {
    const pathname = path.join(fixturesPath, 'insomnia-v4', 'insomnia_v4.yaml');
    const db = await insomniaAdapter(pathname, ['Environment']);

    expect(db?.ApiSpec.length).toBe(0);
    expect(db?.Environment.length).toBe(2);
    expect(db?.Request.length).toBe(0);
    expect(db?.RequestGroup.length).toBe(0);
    expect(db?.Workspace.length).toBe(0);
    expect(db?.UnitTestSuite.length).toBe(0);
    expect(db?.UnitTest.length).toBe(0);
  });

  it('should return null if pathname is invalid', async () => {
    const pathname = path.join(fixturesPath, 'insomnia-v4', 'insomnia');
    const db = await insomniaAdapter(pathname);
    expect(db).toBe(null);
  });
});
