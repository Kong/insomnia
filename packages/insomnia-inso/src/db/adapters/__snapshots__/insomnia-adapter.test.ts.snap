// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insomniaAdapter() should throw InsoError if malformed yaml content: empty.yaml 1`] = `"Failed to parse empty.yaml."`;

exports[`insomniaAdapter() should throw InsoError if malformed yaml content: malformed.yaml 1`] = `"Failed to parse malformed.yaml."`;

exports[`insomniaAdapter() should throw InsoError if malformed yaml content: no-export-format.yaml 1`] = `"Expected an Insomnia v4 export file; unexpected data found in no-export-format.yaml."`;

exports[`insomniaAdapter() should throw InsoError if malformed yaml content: v3-export-format.yaml 1`] = `"Expected an Insomnia v4 export file; found an Insomnia v3 export file in v3-export-format.yaml."`;
